state/remote/swift.go

```diff
@@ -32,50 +50,201 @@ func swiftFactory(conf map[string]string) (Client, error) {
+		if authUrl == "" {
+			return fmt.Errorf("missing 'auth_url' configuration or OS_AUTH_URL environment variable")
+		}
+	}
+	c.authurl = authUrl
+
+	username, ok := conf["user_name"]
+	if !ok {
+		username = os.Getenv("OS_USERNAME")
+	}
+	c.username = username
+
+	userID, ok := conf["user_id"]
+	if !ok {
+		userID = os.Getenv("OS_USER_ID")
+	}
+	c.userid = userID
+
+	password, ok := conf["password"]
+	if !ok {
+		password = os.Getenv("OS_PASSWORD")
+		if password == "" {
+			return fmt.Errorf("missing 'password' configuration or OS_PASSWORD environment variable")
+		}
 	}
-	if val := os.Getenv("OS_USERNAME"); val == "" {
-		return fmt.Errorf("missing OS_USERNAME environment variable")
+	c.password = password
+
+	region, ok := conf["region_name"]
+	if !ok {
+		region = os.Getenv("OS_REGION_NAME")
+		if region == "" {
+			return fmt.Errorf("missing 'region_name' configuration or OS_REGION_NAME environment variable")
+		}
+	}
+	c.region = region
+
+	tenantID, ok := conf["tenant_id"]
+	if !ok {
+		tenantID = os.Getenv("OS_TENANT_ID")
+		if tenantID == "" {
+			tenantID = os.Getenv("OS_PROJECT_ID")
+		}
+	}
+	c.tenantid = tenantID
+
+	tenantName, ok := conf["tenant_name"]
+	if !ok {
+		tenantName = os.Getenv("OS_TENANT_NAME")
+		if tenantName == "" {
+			tenantName = os.Getenv("OS_PROJECT_NAME")
+		}
 	}
-	if val := os.Getenv("OS_TENANT_NAME"); val == "" {
-		return fmt.Errorf("missing OS_TENANT_NAME environment variable")
+	c.tenantname = tenantName
+
+	domainID, ok := conf["domain_id"]
+	if !ok {
+		domainID = os.Getenv("OS_USER_DOMAIN_ID")
+		if domainID == "" {
+			domainID = os.Getenv("OS_PROJECT_DOMAIN_ID")
+			if domainID == "" {
+				domainID = os.Getenv("OS_DOMAIN_ID")
+			}
+		}
 	}
-	if val := os.Getenv("OS_PASSWORD"); val == "" {
-		return fmt.Errorf("missing OS_PASSWORD environment variable")
+	c.domainid = domainID
+
+	domainName, ok := conf["domain_name"]
+	if !ok {
+		domainName = os.Getenv("OS_USER_DOMAIN_NAME")
+		if domainName == "" {
+			domainName = os.Getenv("OS_PROJECT_DOMAIN_NAME")
+			if domainName == "" {
+				domainName = os.Getenv("OS_DOMAIN_NAME")
+				if domainName == "" {
+					domainName = os.Getenv("DEFAULT_DOMAIN")
+				}
+			}
+		}
 	}
+	c.domainname = domainName
+
 	path, ok := conf["path"]
 	if !ok || path == "" {
 		return fmt.Errorf("missing 'path' configuration")
 	}
+	c.path = path
 
-	provider, err := openstack.AuthenticatedClient(gophercloud.AuthOptions{
-		IdentityEndpoint: os.Getenv("OS_AUTH_URL"),
-		Username:         os.Getenv("OS_USERNAME"),
-		TenantName:       os.Getenv("OS_TENANT_NAME"),
-		Password:         os.Getenv("OS_PASSWORD"),
-	})
+	insecure, ok := conf["insecure"]
+	if !ok {
+		insecure = os.Getenv("OS_INSECURE")
+		if insecure == "" {
+			// Default if no conf or Env variable set
+			insecure = "false"
+		}
+	}
+	insecureBool, err := strconv.ParseBool(insecure)
+	if err != nil {
+		return fmt.Errorf("non-boolean 'insecure' value")
+	}
+	c.insecure = insecureBool
```

Comment:
Yeh, I did look at the S3 bool parsing, but have the added complication here of wanting to pull from either a `conf` value or an env var...

Will see if I can tighten up a bit...